float64 x
float64 y
int32 dest_id
---
int32 node_id